### PR TITLE
[Feature][zeta] Add the UNKNOWABLE job status.

### PR DESCRIPTION
**Pull request by ic4y**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java [hunk 1]
```diff
@@ -732,6 +732,7 @@ public void testStreamJobRestoreIn2NodeMasterDown()
     @Disabled
     public void testFor() throws ExecutionException, InterruptedException {
         for (int i = 0; i < 200; i++) {
+            System.out.println("testFor: " + i);
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]: Revert ?

Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceIT.java]:
```suggestion
```

**Review by XiaoJiang521 — APPROVED** [review]

**Comment by XiaoJiang521** [comment]
YYDS

**Review by liugddx — DISMISSED** [review]

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java [hunk 2]
```diff
@@ -232,13 +232,19 @@ public boolean taskIsEnded(@NonNull TaskGroupLocation taskGroupLocation) {
...
-                            true,
-                            exception -> exception instanceof NullPointerException && isRunning,
-                            Constant.OPERATION_RETRY_SLEEP));
+            return Boolean.TRUE.equals(
+                    RetryUtils.retryWithException(
+                            () ->
+                                    nodeEngine.getMasterAddress() != null
+                                            && nodeEngine
+                                                    .getMasterAddress()
+                                                    .equals(nodeEngine.getThisAddress()),
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java]: just `nodeEngine.getThisAddress().equals(nodeEngine.getMasterAddress())`

**Review by liugddx — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**